GlobalSettings.php

```diff
@@ -36,6 +36,35 @@
 	};
 }
 
+// log suspicious or sensitive login attempts
+$wgHooks['AuthManagerLoginAuthenticateAudit'][] = static function ( $response, $user, $username ) {
+	$guessed = false;
+	if ( !$user && $username ) {
+		$user = MediaWikiServices::getInstance()
+			->getUserFactory()
+			->newFromName( $username );
+		$guessed = true;
+	}
+	if ( !$user || !in_array( $response->status,
+		[ AuthenticationResponse::PASS, AuthenticationResponse::FAIL ], true )
+	) {
+		return;
+	}
+	global $wgRequest;
+	$successful = $response->status === AuthenticationResponse::PASS;
+	$channel = $successful ? 'goodpass' : 'badpass';
+	$logger = LoggerFactory::getInstance( $channel );
```

Comment:
[When moved to MirahezeFunctions] needs use statement `use MediaWiki\Logger\LoggerFactory;`